components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -998,6 +1000,22 @@ private Condition getSearchCondition(String templateType, String tenantDomain, S
         return null;
     }
 
+    private void validateJitProvisioningConfig(IdentityProvider identityProvider,
+                                               JustInTimeProvisioning justInTimeProvisioningConfig)
+            throws IdentityProviderManagementException {
+
+        int configuredAppCount =
+                IdentityProviderManager.getInstance().getConnectedApplications(identityProvider.getResourceId(),
+                        null, 0, ContextLoader.getTenantDomainFromContext()).getTotalAppCount();
+        boolean currentProxyModeStatus = justInTimeProvisioningConfig.getIsEnabled();
+        boolean previousProxyModeStatus = identityProvider.getJustInTimeProvisioningConfig().isProvisioningEnabled();
+        if (configuredAppCount > 0 && currentProxyModeStatus != previousProxyModeStatus) {
```

Comment:
does this mean, if there are configured App, the proxy mode can't be disabled also?

Reply:
Yes. when an application is already configured with proxy mode enabled IDP, if we try to disable it, it will affect the expected behaviour of the already configured application.  Therefore to prevent from a situation like that we do not allow to update proxy mode status, if there is an already configured ap with a particular IDP.